best-practices.md

```diff
@@ -0,0 +1,30 @@
+# ML Model Extension Best Practices
+
+This document makes a number of recommendations for creating real world ML Model Extensions. None of them are required to meet the core specification, but following these practices will improve the documentation of your model and make life easier for client tooling and users. They come about from practical experience of implementors and introduce a bit more 'constraint' for those who are creating STAC objects representing their models or creating tools to work with STAC.
+
+## Using STAC Common Metadata Fields for the ML Model Extension
+
+We recommend using the `start_datetime` and `end_datetime`, `geometry`, and `bbox` to represent the recommended context of data the model was trained with and for which the model should have appropriate domain knowledge for inference. For example, we can consider a model which is trained on imagery from all over the world and is robust enough to be applied to any time period. In this case, the common metadata to use with the model would include the bbox of "the world" `[-90, -180, 90, 180]` and the start_datetime and end_datetime range could be generic values like `["1900-01-01", null]`.
```

Comment:
Missing code quotes for `start_datetime` and `end_datetime` toward the end.

I think a counter example could help in this case to get the point across. We do not want to motivate people to always plug "any extent" as the quick fix. In reality, I highly doubt any model works in all ranges (unless they perform very basic classifications, and even then...), only considering the variability of seasons and biodiversity of the terrain. I suggest we even add an explicit mention that "the world" and "any time" definitions should be avoided as much as possible, as it can lead to misuse of the model.

There should be some mention that if a specific extent is known where the model works well (eg: areas for which it was tested with `ml-aoi:split: validation / test` for example), and even more, if some outlier extents **are known** to result in bad predictions, that the extent should indicate the area/time where the model is expected to "work as intended" by excluding the problem/unsupported areas.

Reply:
Taking direct inspiration from https://github.com/stac-extensions/ml-model#spatiotemporal-fields

Reply:
added explanation here, thanks!